src/CMakeLists.txt

```diff
@@ -46,6 +47,8 @@ endif()
 
 add_subdirectory(VecSim/spaces)
 
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall")
```

Comment:
Don't you want to set those in a somewhat more global location? (`build/cmake/common.cmake` or something)
Same note for spaces.

Reply:
no
as this will build the google deps with those flags, and there are warnings there.